src/components/player-dialog.tsx

```diff
@@ -27,86 +36,529 @@ export default function PlayerDialog({
 
   return (
     <Dialog open={isOpen} onOpenChange={onOpenChange}>
-      <DialogContent>
-        <DialogHeader>
-          <DialogTitle>{player.playerName}</DialogTitle>
-        </DialogHeader>
+      <DialogContent className="max-w-md p-0 overflow-hidden rounded-2xl border-0 shadow-lg [&>button]:hidden">
         <VisuallyHidden.Root>
+          <DialogTitle>Player Stats: {player.playerName}</DialogTitle>
           <DialogDescription>Stats for {player.playerName}</DialogDescription>
         </VisuallyHidden.Root>
 
-        <div className="flex flex-col items-center">
-          <Image
-            src={`https://cdn.nba.com/headshots/nba/latest/260x190/${player.playerId}.png`}
-            alt={`${player.playerName} headshot`}
-            className="w-32"
-            width={260}
-            height={190}
-          />
-        </div>
+        <div className="flex flex-col">
+          {/* Header with player info */}
+          <div 
+            className="relative h-56 overflow-hidden"
+            style={{
+              background: headerGradient,
+              marginBottom: '-1px', /* Ensure no gap between elements */
+            }}
+          >
+            {/* Custom close button with improved visibility */}
+            <button
+              type="button"
+              onClick={onClose}
+              className="absolute top-4 right-5 z-50 rounded-full p-2 bg-black/30 text-white hover:bg-black/50 backdrop-blur-sm transition-colors"
+              aria-label="Close"
+            >
+              <X className="h-4 w-4" />
+            </button>
 
-        <div className="flex flex-row flex-wrap justify-center gap-x-2 gap-y-2">
-          <StatBlock label="Pts" value={player.points} />
-          <StatBlock label="Reb" value={player.rebounds} />
-          <StatBlock label="Ast" value={player.assists} />
-          <StatBlock label="Stl" value={player.steals} />
-          <StatBlock label="Blk" value={player.blocks} />
-          <StatBlock label="3pt" value={player.threePointersMade} />
-          <StatBlock
-            label="Fg"
-            value={`${player.fieldGoalsMade}/${player.fieldGoalsAttempted}`}
-          />
-          <StatBlock
-            label="Ft"
-            value={`${player.freeThrowsMade}/${player.freeThrowsAttempted}`}
-          />
-          <StatBlock label="TO" value={player.turnovers} />
-          <StatBlock label="Mins" value={trimmedMinutes} />
-        </div>
+            {/* Team logo watermark */}
+            <div className="absolute inset-0 flex items-center justify-center overflow-hidden pointer-events-none">
+              <div className="opacity-10">
+                <Image 
+                  src={`https://cdn.nba.com/logos/nba/${getTeamIdFromAbbr(playerTeam.abbreviation)}/global/L/logo.svg`}
+                  alt={`${playerTeam.abbreviation} logo`}
+                  width={300}
+                  height={300}
```

Comment:
it looks like these are 150x150 from the source. next will complain if you dont pass the source dimensions here